x-pack/plugins/cloud_security_posture/public/pages/rules/index.tsx

```diff
@@ -46,16 +45,16 @@ export const Rules = ({ match: { params } }: RouteComponentProps<PageUrlParams>)
         ),
       },
     }),
-    [integrationInfo.data, integrationInfo.status]
+    [integrationInfo.data]
   );
 
   return (
-    <CspPageTemplate {...pageProps}>
+    <CspPageTemplate
+      {...pageProps}
+      query={integrationInfo}
+      errorRender={(error) => <RulesErrorPrompt error={extractErrorBodyMessage(error)} />}
+    >
       {integrationInfo.status === 'success' && <RulesContainer />}
```

Comment:
after looking at the code, i think it will ultimately by clearer if we do one of these:
1 - let the template render states
2 - we render states

the mix between them reads odd. i know it's opt in, but in this case i would probably left it as-is so it's clearer how each state gets rendered.

eventually we might be able to share the error prompt, which would mean this entire component could pass the query result.

Reply:
it might be my opinion but having the entire `{status === loading  && <Loading />} {status === error  && <Error />} {status === success  && <Cmp noDataConfig={status !== 'loading' && status !== 'error && noDataObject}/>}` etc etc every page is way too repetitive and really not needed unless a special case arrives. let's roll with this for a bit thats the whole point of the PR..  its just the basic states of the query